### PR TITLE
Fix `zenml stack describe` bug

### PR DESCRIPTION
**Pull request by strickvl**
## Describe changes
Fixed the bug on the `zenml stack describe` command caused by a change in what was passed into the print/output function.
Added a test to check for that going forward.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/features/integrations) table.
- [x] I have added tests to cover my changes.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: tests/unit/cli/test_utils.py [hunk 1]
```diff
@@ -41,3 +42,17 @@ def test_parse_unknown_options_returns_a_dict_of_known_options() -> None:
     assert isinstance(parsed_sample_args, dict)
     assert len(parsed_sample_args.values()) == 3
     assert parsed_sample_args["best_cat"] == '"aria"'
+
+
+def test_stack_config_has_right_contents_for_printing() -> None:
+    """Check that the stack config has the right components for printing"""
+    repo = Repository()
```
Comment (htahir1) [inline tests/unit/cli/test_utils.py:49]: Is this how we access the repository now in our tests? I am for sure outdated on the details but I thought we used to do this via pytest fixtures?

File: src/zenml/steps/base_step.py [hunk 2]
```diff
@@ -592,7 +592,7 @@ def __call__(
         )
 
         self.INPUT_SPEC = {
-            arg_name: artifact_type.type  # type:ignore[misc]
```
Comment (htahir1) [inline src/zenml/steps/base_step.py:595]: hmm i hope mypy doesnt complain

**Review by htahir1 — COMMENTED** [review]
Ok looks good! Just a few comments

File: src/zenml/steps/base_step.py (hunk 2 above)
Comment (strickvl) [reply src/zenml/steps/base_step.py:595]: This somehow was in develop when I first branched off it. `mypy` complained that there was an unused `type:ignore` comment and told me to remove it. No idea how that got merged to `develop` in the first place.

File: tests/unit/cli/test_utils.py (hunk 1 above)
Comment (strickvl) [reply tests/unit/cli/test_utils.py:49]: Checked and `zenml/tests/unit/test_repository.py` and others like `zenml/tests/unit/orchestrators/test_context_utils.py` are using `Repository()` so I assume that's what we're doing?

**Comment by strickvl** [comment]
@stefannica adding you on this as you said you've already fixed this.

File: src/zenml/cli/utils.py [hunk 3]
```diff
@@ -181,7 +181,7 @@ def print_stack_configuration(
     )
     rich_table.add_column("COMPONENT_TYPE")
     rich_table.add_column("COMPONENT_NAME")
-    items = {typ.value: name for typ, name in config.items()}
+    items = [[typ.value, name] for typ, name in config.items()]
```
Comment (stefannica) [inline src/zenml/cli/utils.py:184]: my version of it was this: ```    items = ([typ.value, name] for typ, name in config.items())
    for item in items:
        rich_table.add_row(*item)```

I think you can remove the `*list(item)` part too.

Comment (strickvl) [reply src/zenml/cli/utils.py:184]: Even better. Thank you. I've updated it with how you did it.

**Review by stefannica — APPROVED** [review]

**Review by htahir1 — APPROVED** [review]
LGTm!

**Pull request merged**